src/offset/local/unix.rs

```diff
@@ -8,64 +8,118 @@
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
 
-use std::sync::Once;
+use std::{cell::RefCell, env, fs, time::SystemTime};
 
 use super::tz_info::TimeZone;
 use super::{DateTime, FixedOffset, Local, NaiveDateTime};
 use crate::{Datelike, LocalResult, Utc};
 
 pub(super) fn now() -> DateTime<Local> {
     let now = Utc::now().naive_utc();
-    DateTime::from_utc(now, offset(now, false).unwrap())
+    naive_to_local(&now, false).unwrap()
 }
 
 pub(super) fn naive_to_local(d: &NaiveDateTime, local: bool) -> LocalResult<DateTime<Local>> {
-    if local {
-        match offset(*d, true) {
-            LocalResult::None => LocalResult::None,
-            LocalResult::Ambiguous(early, late) => LocalResult::Ambiguous(
-                DateTime::from_utc(*d - early, early),
-                DateTime::from_utc(*d - late, late),
+    TZ_INFO.with(|maybe_cache| {
+        // can later change `.replace(None)` with  `.take()` if MSRV goes >= 1.50
+        // can also remove the `ref` if/when MSRV increases to allow mixed owned/ref patterns
+        let mut cache = maybe_cache.replace(None).unwrap_or_default();
```

Comment:
Seems like a case for [`Option::get_or_insert()`](https://doc.rust-lang.org/std/option/enum.Option.html#method.get_or_insert)?